### DISCUSSION TITLE
Indian accent for English not getting identified correctly

### DISCUSSION THREAD

**Discussion by RRThivyan** [speechbrain/speechbrain#2694] [Q&A]
Indian accent for English not getting identified correctly. Other Indian languages are identified correctly but English in Indian accent is not read correctly. It says its Tamil, Hindi, Telugu etc but not English. But it reads other English accents correctly sans issue.

**Comment by asumagic** [comment]
You haven't specified which model you tried. This likely just comes down to which datasets the model was trained on.

**Reply by RRThivyan** [reply]
Hi, I am using speechbrain/lang-id-voxlingua107-ecapa to identify the language of the audio input, The model is not able to identify the Indian accent English but identify other English accents. Is there any documentation to finetune the model?

I am using a dataset from huggingface for Indian accent English and modified the naming convention as similar to the one from the link https://bark.phon.ioc.ee/voxlingua107/ but getting the following error. From my understanding, its an issue with the target labels used in the loss computation

![image](https://github.com/user-attachments/assets/b1f6728e-06c2-494b-bc67-18adf64e61cb)


![image](https://github.com/user-attachments/assets/21169c67-83cf-4eea-81e9-bf4ac52373ef)

after using CUDA_LAUNCH_BLOCKING=1 python train.py hparams/train_ecapa.yaml

![image](https://github.com/user-attachments/assets/ad58a402-1bcc-4d72-b4c4-46ebf62d8e01)
